src/main/scala/xerial/sbt/sonatype/SonatypeException.scala

```diff
@@ -20,7 +22,18 @@ object SonatypeException {
 
   case object MISSING_STAGING_PROFILE extends ErrorCode
 
-  case object MISSING_PROFILE extends ErrorCode
+  case class MISSING_PROFILE(profileName: String, host: String) extends ErrorCode {
+    val problem = s"Profile $profileName is not found on $host"
+
+    val possibleAlternativeHosts: Seq[String] = Sonatype.KnownOssHosts.filterNot(_ == host)
+
+    val hostAdvice = s"try ${possibleAlternativeHosts.mkString(", or ")}?"
+
+    val advice: String =
+      s"In your sbt settings, check your sonatypeProfileName and sonatypeCredentialHost ($hostAdvice)"
+
+    val message: String = s"$problem. $advice"
+  }
```

Comment:
Changing `MISSING_PROFILE` into a case class lets us break down the construction of the error message into separate fields, and makes it easier to construct good assertions on the error messages in `SonatypeExceptionTest`.